File: merge-strings-alternately/MergeStrings.java
public class MergeStrings {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder merged = new StringBuilder();
        int i = 0;
        while (i<word1.length() || i<word2.length()) {
            if (i<word1.length()){
                merged.append(word1.charAt(i));
            }
            if (i<word2.length()){
                merged.append(word2.charAt(i));
            }
            i++;
        }
        return merged.toString();
}
public static void main(String[] args) {
    MergeStrings m = new MergeStrings();
    String result = m.mergeAlternately("koira", "maukkakissa");
    System.out.println(result);
}
}

File: merge-strings-alternately/merge-strings.py
# Leetcode 1768. Merge Strings Alternately
#https://leetcode.com/problems/merge-strings-alternately/

def mergeAlternately(word1: str, word2: str)-> str:
    merged=[]
    i=0
    while i<len(word1) or i<len(word2):
        if i>=len(word1):
            merged.append(word2[i])
        elif i<len(word1) and i<len(word2):           
            merged.append(word1[i])
            merged.append(word2[i])
        elif i>=len(word2):
            merged.append(word1[i])
        i+=1          
    return "".join(merged)
if __name__=="__main__":
    print(mergeAlternately("koira","maukkakissa"))
